homeassistant/components/blink/__init__.py

```diff
@@ -164,13 +184,33 @@ async def async_handle_save_video_service(hass, entry, call):
         _LOGGER.error("Can't write %s, no access to path!", video_path)
         return
 
-    def _write_video(camera_name, video_path):
+    def _write_video(name, file_path):
         """Call video write."""
         all_cameras = hass.data[DOMAIN][entry.entry_id].cameras
-        if camera_name in all_cameras:
-            all_cameras[camera_name].video_to_file(video_path)
+        if name in all_cameras:
+            all_cameras[name].video_to_file(file_path)
 
     try:
         await hass.async_add_executor_job(_write_video, camera_name, video_path)
     except OSError as err:
         _LOGGER.error("Can't write image to file: %s", err)
+
+
+async def async_handle_save_recent_clips_service(hass, entry, call):
+    """Save multiple recent clips to output directory."""
+    camera_name = call.data[CONF_NAME]
+    clips_dir = call.data[CONF_FILE_PATH]
+    if not hass.config.is_allowed_path(clips_dir):
+        _LOGGER.error("Can't write to directory %s, no access to path!", clips_dir)
+        return
+
+    def _save_recent_clips(name, output_dir):
```

Comment:
That is the crux of the PR.  The `_write_video` only triggers a single video (the most recent) to be written.  I left it in so that current users' automations do not break.

The new "save recent clips" adds a service for saving all clips generated since the last refresh.

Reply:
As far as your question about access to `hass`.  I followed the same pattern as the `async_handle_save_video_service` function.  I'm not trying to do a refactor here.

Reply:
thanks for clarifying